bzlib.rs

```diff
@@ -2889,7 +2884,8 @@ pub unsafe extern "C" fn BZ2_bzclose(b: *mut BZFILE) {
         }
     }
 
-    if fp != stdin && fp != stdout {
+    // don't close stdin or stdout
+    if libc::fileno(fp) > 1 {
```

Comment:
This should probably compare against 0 and 1. If the `FILE` is not associated with a file on the disk, but a buffer in memory, `fileno` will return `-1` and set `errno`, yet we would still want to fclose the `FILE`.

Reply:
right, although as the C snippet above shows, file descriptor 1 could point to an actual file too (which we'd want to close).

Reply:
> file descriptor 1 could point to an actual file too (which we'd want to close).

No, you don't ever want to manually close fd 0/1/2 unless you are in a single threaded context and immediately replace them with a new open file. Other than that you should let the OS implicitly close it at process exit. A lot of code out there assumes that fd 0/1/2 always exist (libstd even opens them as `/dev/null` at startup if they don't exist)